presto-main/src/main/java/io/prestosql/sql/planner/LocalExecutionPlanner.java

```diff
@@ -950,20 +953,43 @@ public PhysicalOperation visitWindow(WindowNode node, LocalExecutionPlanContext
             ImmutableList.Builder<Symbol> windowFunctionOutputSymbolsBuilder = ImmutableList.builder();
             for (Map.Entry<Symbol, WindowNode.Function> entry : node.getWindowFunctions().entrySet()) {
                 Optional<Integer> frameStartChannel = Optional.empty();
+                Optional<Integer> sortKeyChannelForStartComparison = Optional.empty();
                 Optional<Integer> frameEndChannel = Optional.empty();
+                Optional<Integer> sortKeyChannelForEndComparison = Optional.empty();
+                Optional<Integer> sortKeyChannel = Optional.empty();
+                Optional<Ordering> ordering = Optional.empty();
 
                 Frame frame = entry.getValue().getFrame();
                 if (frame.getStartValue().isPresent()) {
                     frameStartChannel = Optional.of(source.getLayout().get(frame.getStartValue().get()));
                 }
+                if (frame.getSortKeyCoercedForFrameStartComparison().isPresent()) {
+                    sortKeyChannelForStartComparison = Optional.of(source.getLayout().get(frame.getSortKeyCoercedForFrameStartComparison().get()));
+                }
                 if (frame.getEndValue().isPresent()) {
                     frameEndChannel = Optional.of(source.getLayout().get(frame.getEndValue().get()));
                 }
+                if (frame.getSortKeyCoercedForFrameEndComparison().isPresent()) {
+                    sortKeyChannelForEndComparison = Optional.of(source.getLayout().get(frame.getSortKeyCoercedForFrameEndComparison().get()));
+                }
+                if (node.getOrderingScheme().isPresent()) {
+                    sortKeyChannel = Optional.of(sortChannels.get(0));
```

Comment:
sortChannels is a list. why using the first element only is sufficient?
or is the list known to be single element in this branch?

Reply:
`sortKeyChannel` and `ordering` are only used for window frame using `RANGE PRECEDING` or `RANGE FOLLOWING`. With such frame definition, a single sort item is required per spec, and it is guaranteed by the Analyzer.

Reply:
That wouldn't work. In general, we can have multiple sort channels at this point. But for frame type RANGE, there is only one. We capture it here and use later only for RANGE frames.

I added a clarifying comment: https://github.com/trinodb/trino/pull/24157.

Reply:
What if it has multiple sortkeyChannel for frame type RANGE - Would we skip them ?